test/runAll.unmocked.spec.js

```diff
@@ -370,6 +325,69 @@ index f80f875..1c5643c 100644
     expect(await readFile('test.js')).toEqual(testJsFileUgly + appended)
   })
 
+  it('should handle merge conflicts', async () => {
+    const fileInBranchA = `module.exports = "foo";\n`
+    const fileInBranchB = `module.exports = 'bar'\n`
+    const fileInBranchBFixed = `module.exports = "bar";\n`
+
+    // Create one branch
+    await execGit(['checkout', '-b', 'branch-a'])
+    await appendFile('test.js', fileInBranchA)
+    await execGit(['add', '.'])
+    const successA = await gitCommit(fixJsConfig, 'commit a')
+    expect(successA).toEqual(true)
+    expect(await readFile('test.js')).toEqual(fileInBranchA)
+
+    await execGit(['checkout', 'master'])
+
+    // Create another branch
+    await execGit(['checkout', '-b', 'branch-b'])
+    await appendFile('test.js', fileInBranchB)
+    await execGit(['add', '.'])
+    const successB = await gitCommit(fixJsConfig, 'commit b')
+    expect(successB).toEqual(true)
+    expect(await readFile('test.js')).toEqual(fileInBranchBFixed)
+
+    // Merge first branch
+    await execGit(['checkout', 'master'])
+    await execGit(['merge', 'branch-a'])
+    expect(await readFile('test.js')).toEqual(fileInBranchA)
+    expect(await execGit(['log', '-1', '--pretty=%B'])).toMatch('commit a')
+
+    // Merge second branch, causing merge conflict
+    try {
+      await execGit(['merge', 'branch-b'])
+    } catch ({ stdout }) {
+      expect(stdout).toMatch('Merge conflict in test.js')
+    }
+
+    expect(await readFile('test.js')).toMatchInlineSnapshot(`
+      "<<<<<<< HEAD
+      module.exports = \\"foo\\";
+      =======
+      module.exports = \\"bar\\";
+      >>>>>>> branch-b
+      "
+    `)
+
+    // Fix conflict and commit using lint-staged
+    await writeFile('test.js', fileInBranchB)
+    expect(await readFile('test.js')).toEqual(fileInBranchB)
+    await execGit(['add', '.'])
+
+    // Do not use `gitCommit` wrapper here
+    await runAll({ ...fixJsConfig, cwd, quiet: true })
+
+    // Lint-staged lost MERGE_HEAD
+    try {
+      await execGit(['merge', '--continue'])
+    } catch ({ stderr }) {
+      expect(stderr).toMatch('There is no merge in progress (MERGE_HEAD missing)')
```

Comment:
This behaviour is incorrect, but I made the test check for it. I should fix this before merging.